static/app/views/replays/replayTable.tsx

```diff
@@ -59,113 +53,95 @@ function ReplayTable({replayList, idKey, showProjectColumn}: Props) {
         'count_if(event.type,equals,error)',
       ],
       orderby: '-min_timestamp',
-      environment: selection.environments,
-      projects: selection.projects,
       query: `(title:"sentry-replay-event-*" OR event.type:error) AND (${query})`,
-    };
+    }),
+    [query]
+  );
+
+  const {data} = useDiscoverQuery<ReplayDurationAndErrors>({discoverQuery});
 
-    if (selection.datetime.period) {
-      eventQueryParams.range = selection.datetime.period;
-    }
-    return EventView.fromNewQueryWithLocation(eventQueryParams, location);
-  };
+  const dataEntries = data
+    ? Object.fromEntries(data.map(item => [item.replayId, item]))
+    : {};
 
   return (
-    <DiscoverQuery
-      eventView={getEventView()}
-      location={location}
-      orgSlug={organization.slug}
-    >
-      {data => {
-        const dataEntries = data.tableData
-          ? Object.fromEntries(
-              (data.tableData?.data as ReplayDurationAndErrors[]).map(item => [
-                item.replayId,
-                item,
-              ])
-            )
-          : {};
-        return replayList?.map(replay => {
-          return (
-            <Fragment key={replay.id}>
-              <UserBadge
-                avatarSize={32}
-                displayName={
-                  <Link
-                    to={`/organizations/${organization.slug}/replays/${generateEventSlug({
-                      project: replay.project,
-                      id: replay[idKey],
-                    })}/`}
-                  >
-                    {replay['user.display']}
-                  </Link>
+    <Fragment>
```

Comment:
Is this Fragment needed? you should be able to return the `replayList.map` directly

Reply:
Was thinking about that `Fragment`, one of the first options was not send it, but Typescript is complaining about this:

`'ReplayTable' cannot be used as a JSX component.
  Its return type 'Element[]' is not a valid JSX element.`